server/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java

```diff
@@ -219,24 +225,27 @@ private static void initPhase2(Bootstrap bootstrap) throws IOException {
         // load the plugin Java modules and layers now for use in entitlements
         var modulesBundles = PluginsLoader.loadModulesBundles(nodeEnv.modulesFile());
         var pluginsBundles = PluginsLoader.loadPluginsBundles(nodeEnv.pluginsFile());
-        var pluginsLoader = PluginsLoader.createPluginsLoader(modulesBundles, pluginsBundles);
-        bootstrap.setPluginsLoader(pluginsLoader);
+
+        final PluginsLoader pluginsLoader;
 
         if (bootstrap.useEntitlements()) {
             LogManager.getLogger(Elasticsearch.class).info("Bootstrapping Entitlements");
 
             var pluginData = Stream.concat(
-                pluginsLoader.moduleBundles()
-                    .stream()
+                modulesBundles.stream()
                     .map(bundle -> new PolicyParserUtils.PluginData(bundle.getDir(), bundle.pluginDescriptor().isModular(), false)),
-                pluginsLoader.pluginBundles()
-                    .stream()
+                pluginsBundles.stream()
                     .map(bundle -> new PolicyParserUtils.PluginData(bundle.getDir(), bundle.pluginDescriptor().isModular(), true))
             ).toList();
             var pluginPolicies = PolicyParserUtils.createPluginPolicies(pluginData);
+
+            pluginsLoader = PluginsLoader.createPluginsLoader(modulesBundles, pluginsBundles, findPluginsWithNativeAccess(pluginPolicies));
```

Comment:
Nit: the only difference between these `createPluginsLoader` calls is the third argument. Could we pull that out into a variable, rather than `pluginsLoader` itself, and then have a single call to `createPluginsLoader` at the bottom?

Reply:
Unfortunately this is not possible; we need the pluginsLoader at the very next step, inside this branch.
The alternative is to duplicate the if; I preferred this way (and it's going to be temporary anyway, until the SM branch goes away)